aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -106,6 +108,44 @@ Tensor isinf(const Tensor &self) {
   });
 }
 
+static inline TensorIterator build_is_infinity_op_iterator(Tensor& result, const Tensor& self) {
+  return at::TensorIteratorConfig()
+    .check_all_same_dtype(false)
+    .set_check_mem_overlap(true)
+    .add_output(result)
+    .add_input(self)
+    .cast_common_dtype_to_outputs(true)
```

Comment:
I don't think you need this since you're manually casting from the input type to bool?

Reply:
The `cast_common_dtype_to_outputs(true)` is still needed, as we need to cast the checking result(`bool`) to the dtype of the output tensor.

NumPy supports passing non-bool array to `out` parameters.
```python
In [1]: import numpy as np
In [2]: x = np.array([float('inf'), float('inf')])
In [3]: y = np.array([10, 20])
In [4]: np.isposinf(x, out=y)
Out[4]: array([1, 1])
In [5]: y
Out[5]: array([1, 1])
```

This PR mimics the familiar behaviors
```python
In [1]: import torch
In [2]: x = torch.tensor([float('inf'), float('inf')])
In [3]: y = torch.tensor([10, 20])
In [4]: torch.isposinf(x, out=y)
Out[4]: tensor([1, 1])
In [5]: y
Out[5]: tensor([1, 1])
```